Source/Storages/Realm/Extensions/RealmContext+StorageContext.swift

```diff
@@ -40,7 +40,7 @@ extension RealmContext {
 
             entities.lazy
                 .flatMap { return $0 as? Object }
-                .forEach { realm.delete($0) }
+                .forEach { realm.delete($0, cascading: true) }
```

Comment:
Should we allow the clients to inject this boolean values? May we have some scenarios where we don't want a cascading deletion?